scientific_library/tvb/simulator/simulator.py

```diff
@@ -425,8 +425,9 @@ def _update_and_bound_history(self, history):
 
     def update_state(self, state, node_coupling, local_coupling=0.0):
         # If there are non-state variables, they need to be updated for the initial condition:
-        state = \
-                self.model.update_non_state_variables(state, node_coupling, local_coupling, use_numba=self.use_numba)
+        if self.model._update_non_state_variables:
+            state = \
```

Comment:
Yes, this could be the case. The default implementation is to do nothing. In order to avoid a  redundant function call, I thought to have an if statement inside the loop that checks about whether there is need for any specific model to call this method...

Reply:
As for avoiding numba, I meant at runtime.

Reply:
For the moment, I removed the redundant if statement (there is one already in the __call__ method